pyteal/ast/scratch.py

```diff
@@ -0,0 +1,80 @@
+from ..types import TealType
+from .expr import Expr
+
+class ScratchSlot:
```

Comment:
Is it possible to store type information along with the slot?

Reply:
Perhaps this would add a restriction to the set of programs that you can write in PyTEAL in that a single slot number can't be used for data of two different types statically (even though the program would run fine dynamically due to e.g., branching).

Reply:
Currently the only place slots are used is in `MaybeValue`, which stores the type information. The type is then passed into `ScratchSlot.load()` so that type checking can be done on the loaded value. I think this is acceptable for now, especially because I'm not sure how slot reuse will be implemented in the future.